### PR TITLE
Drop v1beta1 crds for k8s 122 compliance

### PR DESCRIPTION
**Pull request by thomasschoeftner**
### What changes were proposed in this pull request?
I've tried to get rid of `v1beta1` CRD in the Cloudflow code base.

### Why are the changes needed?
With Kubernetes 1.22 support for `v1beta1` CRDs is removed and Cloudflow will not be deployable anymore.
The issue had already been raised by another user: https://github.com/lightbend/cloudflow/issues/1199

### Does this PR introduce any user-facing change?
no

### How was this patch tested?
I do not have an appropriate environment at the moment to verify the change.

### CODE REVIEW

File: core/cloudflow-crd/src/main/scala/akka/datap/crd/App.scala [hunk 1]
```diff
@@ -50,6 +52,15 @@ object App {
   final val ProtocolVersionKey = "protocol-version"
   final val ProtocolVersion = "7"
 
+  val customResourceDefinitionVersion: CustomResourceDefinitionVersion = {
+    new CustomResourceDefinitionVersionBuilder()
+      .withName(GroupVersion)
```
Comment (thomasschoeftner) [inline core/cloudflow-crd/src/main/scala/akka/datap/crd/App.scala:57]: I'm not sure if `name` is the correct property here.

File: core/cloudflow-crd/src/main/scala/akka/datap/crd/App.scala [hunk 2]
```diff
@@ -73,13 +84,9 @@ object App {
       .withPlural(Plural)
       .withShortNames(Short)
       .endNames()
-      .withVersion(GroupVersion)
+      .withVersions(customResourceDefinitionVersion)
       .withScope("Namespaced")
       .withPreserveUnknownFields(true)
-      .withNewSubresources()
```
Comment (thomasschoeftner) [inline core/cloudflow-crd/src/main/scala/akka/datap/crd/App.scala:79]: all these seem to be part of `CustomResourceDefinitionVersion`/ `CustomResourceDefinitionVersionBuilder` now

File: core/cloudflow-operator/src/main/scala/cloudflow/operator/Main.scala [hunk 3]
```diff
@@ -129,11 +129,11 @@ object Main extends {
     Option(
       client
         .apiextensions()
-        .v1beta1()
+        .v1()
         .customResourceDefinitions()
         .withName(App.ResourceName)
         .get()) match {
-      case Some(crd) if crd.getSpec.getVersion == App.GroupVersion =>
+      case Some(crd) if crd.getSpec.getVersions().contains(App.GroupVersion) =>
```
Comment (thomasschoeftner) [inline core/cloudflow-operator/src/main/scala/cloudflow/operator/Main.scala:136]: This match is "weaker" now... not as accurate.
Should we match for "contains only"?

**Comment by thomasschoeftner** [comment]
@RayRoestenburg - any chance you could take a look, pls.

**Comment by franciscolopezsancho** [comment]
I'll have a look at it @thomasschoeftner

File: core/cloudflow-operator/src/main/scala/cloudflow/operator/Main.scala (hunk 3 above)
Comment (thomasschoeftner) [reply core/cloudflow-operator/src/main/scala/cloudflow/operator/Main.scala]: was incorrect -> now checking if `getVersions` contains a `CustomResourceDefinitionVersion` with the appropriate name.

**Comment by thomasschoeftner** [comment]
> I'll have a look at it @thomasschoeftner

Thank you! - I'm not sure whether the changes in `core/cloudflow-crd/src/main/scala/akka/datap/crd/App.scala` are too excessive, but IMO they make sense...

**Review by scpcy — APPROVED** [review]

File: core/cloudflow-crd/src/main/scala/akka/datap/crd/App.scala [hunk 4]
```diff
@@ -50,6 +52,15 @@ object App {
   final val ProtocolVersionKey = "protocol-version"
   final val ProtocolVersion = "7"
 
+  val customResourceDefinitionVersion: CustomResourceDefinitionVersion = {
+    new CustomResourceDefinitionVersionBuilder()
+      .withName(GroupVersion)
+      .withNewSubresources()
+      .withNewStatus()
+      .endStatus()
```
Comment (franciscolopezsancho) [inline core/cloudflow-crd/src/main/scala/akka/datap/crd/App.scala:60]: Hi @thomasschoeftner. Many thanks for the PR. Quick question, why is this `endStatus` needed? or the next `endSubresources`

Comment (franciscolopezsancho) [reply core/cloudflow-crd/src/main/scala/akka/datap/crd/App.scala:60]: I see now what it is. No worries.

**Review by franciscolopezsancho — APPROVED** [review]
LGTM

**Comment by thomasschoeftner** [comment]
thanks @franciscolopezsancho 

I think now these issues are obsolete:
- https://github.com/lightbend/cloudflow/issues/1226
- https://github.com/lightbend/cloudflow/issues/1214
- https://github.com/lightbend/cloudflow/issues/1199

Can you close them, pls?

**Pull request merged**